src/ids/quickId.ts

```diff
@@ -60,6 +68,24 @@ async function specialQuickId(
     }
     return lastDocsRef.ids;
   }
+  const match = quickString.match(_RECENT_REGEX);
+  if (match) {
+    const [, underscores, numberStr, fieldName] = match;
+    let position = underscores.length;
+
+    if (numberStr) {
+      position = parseInt(numberStr, 10);
+      return getRecentDocumentByPosition(db, position, fieldName || null);
+    }
+
+    if (underscores.length === 1 && /^\d+$/.test(fieldName)) {
```

Comment:
this is messy here--fieldName is actually the number string which is confusing. This results because of the noncaptured group in the regex not occurring when there is no number.

To fix this please use named capture groups in your regex: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group

then you can pull out based on name rather than group index which will be both more robust and more readable.